lib/Server.js

```diff
@@ -67,6 +67,16 @@ class Server {
 
     this.log = _log || createLogger(options);
 
+    if (this.options.serverMode === undefined) {
+      this.options.serverMode = 'sockjs';
+    } else {
+      this.log.warn(
+        'serverMode is an experimental option, meaning its usage could potentially change without warning'
+      );
+    }
+
+    this.ServerImplementation = getSocketServerImplementation(this.options);
```

Comment:
I think we should rename variable to `this.socketServerImplementation` to avoid misleading, we already store value in `this.socketServer` so let's use `socket` prefix too

Reply:
I'm going to capitalize first letter to clarify that it is a class.